src/hotspot/share/compiler/compilationMemoryStatistic.cpp

```diff
@@ -405,7 +405,7 @@ void CompilationMemoryStatistic::on_end_compilation() {
   if (env) {
     const char* const failure_reason = env->failure_reason();
     if (failure_reason != nullptr) {
-      result = (failure_reason == failure_reason_memlimit()) ? "oom" : "err";
+      result = (strcmp(failure_reason, failure_reason_memlimit()) == 0) ? "oom" : "err";
```

Comment:
Note: Now that `ciEnv` makes a local copy, we cannot just compare pointers, we need to compare the content.